packages/cadl-rlc-emitter/src/transform/transformPaths.ts

```diff
@@ -19,18 +22,35 @@ export function transformPaths(program: Program): Paths {
   const [routes, _diagnostics] = getAllRoutes(program);
   const paths: Paths = {};
   for (const route of routes) {
+    const parameterBaseName = getParameterBaseName(
+      route.groupName,
+      route.operation.name
+    );
+    const respNames = [];
+    for (const resp of route.responses) {
+      const respBaseName = getResponseBaseName(
+        route.groupName,
+        route.operation.name,
+        resp.statusCode === "*" ? "Default" : resp.statusCode
+      );
+      const respName = getResponseTypeName(respBaseName);
+      respNames.push(respName);
+    }
     const method = {
       description: getDoc(program, route.operation) ?? "",
       hasOptionalOptions: route.parameters.parameters.some(
         (p) => p.param.optional
       ),
-      optionsName: "options",
+      optionsName: getParameterTypeName(parameterBaseName),
```

Comment:
Again if you don't reuse the parameter `parameterBaseName` you can call this way:

```suggestion
      optionsName: getParameterTypeName(
      route.groupName,
      route.operation.name),
```